### PR TITLE
OS signals + Graceful shutdown

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

- Add close method to event consumer allowing the consuming loop
to be exited first.
- Add graceful shutdown of all resources using a timeout context. 
- Update to use new go-ns kafka consumer / producer with blocking close
function. 
- Add default HTTP healthcheck endpoint and added it into the shutdown process

### How to review

Code review + test

### Who can review

Anyone

### CODE REVIEW

File: cmd/dp-dataset-exporter/main.go [hunk 1]
```diff
@@ -85,22 +89,69 @@ func main() {
...
+		err = eventConsumer.Close(ctx)
+		logIfError(err)
 
-	err = dbConnection.Close()
-	if err != nil {
-		log.Error(err, log.Data{"message": "failed to close connection to Neo4j"})
+		err = kafkaConsumer.Close(ctx)
+		logIfError(err)
+
+		err = kafkaProducer.Close(ctx)
```
Comment (nshumoogum) [inline cmd/dp-dataset-exporter/main.go:106]: Should we not close our producers after shutting down server.

For instance, if a kafka message is to be written due to an ongoing request then it wont be written in the current ordering of closing connections to either servers or kafka.

File: event/consumer_test.go [hunk 2]
```diff
@@ -125,6 +132,38 @@ func TestConsume_HandlerError(t *testing.T) {
...
+
+		consumer := event.NewConsumer()
+
+		consumer.Consume(mockConsumer, mockEventHandler, nil)
+
+		Convey("When close is called", func() {
+
+			err := consumer.Close(context.Background())
+
+			Convey("The no errors are returned", func() {
```
Comment (nshumoogum) [inline event/consumer_test.go]: Missing `n` on the end of `The` ;)

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: cmd/dp-dataset-exporter/main.go (hunk 1 above)
Comment (CarlHembrough) [reply cmd/dp-dataset-exporter/main.go:106]: Any existing requests should be complete by the time eventConsumer.Close(ctx) returns, so there should be no issues with the order of closing Kafka resources.

File: event/consumer_test.go (hunk 2 above)
Comment (CarlHembrough) [reply event/consumer_test.go]: done

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**